cypress/App.jsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from 'react';
+import 'bulma/css/bulma.css';
+import './App.scss';
+
+export const goodsFromServer = [
+  'Dumplings',
+  'Carrot',
+  'Eggs',
+  'Ice cream',
+  'Apple',
+  'Bread',
+  'Fish',
+  'Honey',
+  'Jam',
+  'Garlic',
+];
+
+const prepareGoods = (goods, { field, isReverse }) => {
+  const sortedGoods = [...goods];
+
+  if (field === 'alphabetical') {
+    sortedGoods.sort();
+  } else if (field === 'length') {
+    sortedGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isReverse) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortField, setSortField] = useState(null);
+  const [originalGoods] = useState(goodsFromServer);
+
+  const handleSortByName = () => {
+    setSortField('alphabetical');
+    setVisibleGoods(
+      prepareGoods(goodsFromServer, {
+        field: 'alphabetical',
+        isReverse: isReversed,
+      }),
+    );
+  };
+
+  const handleSortByLength = () => {
+    setSortField('length');
+    setVisibleGoods(
+      prepareGoods(goodsFromServer, { field: 'length', isReverse: isReversed }),
+    );
+  };
+
+  const handleReverse = () => {
+    setIsReversed(!isReversed);
+    setVisibleGoods(
+      prepareGoods(goodsFromServer, {
+        field: sortField,
+        isReverse: !isReversed,
+      }),
+    );
+  };
+
+  const handleReset = () => {
+    setVisibleGoods(originalGoods);
```

Comment:
The reset functionality sets visibleGoods to originalGoods, but since originalGoods is initialized with goodsFromServer, it may not reflect the original order if sorting or reversing has been applied. Consider resetting visibleGoods to goodsFromServer directly or maintaining a separate state for the original order.